sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+```java
+// specify token properties
+ServiceSASSignatureValues value = new ServiceSASSignatureValues()
+BlobSASPermission p = new BlobSASPermission()
+        .withRead(true)
+        .withWrite(true)
+        .withCreate(true)
+        .withDelete(true)
+        .withAdd(true)
+value.withPermissions(p.toString())
+        .withStartTime(OffsetDateTime.now().minusDays(1))
+        .withExpiryTime(OffsetDateTime.now().plusDays(1))
+        .withContainerName(containerName)
+        .withBlobName(blobName)
+IPRange ipR = new IPRange()
+        .withIpMin("0.0.0.0")
+        .withIpMax("255.255.255.255")
+value.withIpRange(ipR)
+        .withProtocol(SASProtocol.HTTPS_ONLY)
+        .withCacheControl("cache")
+        .withContentDisposition("disposition")
+        .withContentEncoding("encoding")
+        .withContentLanguage("language")
+        .withContentType("type")
+
+// build the token
+SASQueryParameters sasToken = value.generateSASQueryParameters(sharedKeyCredentails)
+```
+
+In v12, SAS tokens are generated off the client to the resource you wish to generate the SAS for.
+The resource path handled automatically because of this, and the `SharedKeyCredential` authenticating the client is also used automatically to sign the SAS. See [building a client](#buildclient) for how to get an authenticated client.
+```java
+// specify token properties
+String identifier = "identifier";
+BlobSASPermission permissions = new BlobSASPermission()
+        .read(true)
+        .create(true)
+        .delete(true)
+        .write(true);
+OffsetDateTime startTime = OffsetDateTime.now().minusDays(1);
+OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+IPRange ipRange = new IPRange()
+        .ipMin("0.0.0.0")
+        .ipMax("255.255.255.255");
+SASProtocol sasProtocol = SASProtocol.HTTPS_HTTP;
+String cacheControl = "cache";
+String contentDisposition = "disposition";
+String contentEncoding = "encoding";
+String contentLanguage = "language";
+String contentType = "type";
+String version = Constants.HeaderConstants.TARGET_STORAGE_VERSION;
+
+// build the token
+String sasToken = blobClient.generateSAS(identifier, permissions, expiryTime, startTime, version, sasProtocol,
+            ipRange, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType);
+```
+
+## Build Client
+In v10/v11, core classes were built with constructors, taking a `URL` and an already constructed `HttpPipeline`.
+```java
+final BlockBlobURL blobURL = new BlockBlobURL(
+    new URL("https://" + accountName + ".blob.core.windows.net/mycontainer/myimage.jpg"), 
+    StorageURL.createPipeline(myCredential, new PipelineOptions()));
+```
+
+In v12, we have moved to a builder pattern. To replicate the above v10/v11 snippit:
+```java
+final BlockBlobClient blobClient = new BlobClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net")
```

Comment:
I think this would be a more direct comparison if we use the same endpoint with the container name and blob name instead of setting them separately. That way customers don't think they have to separate them out before using the builder.